### PR TITLE
elvish: update to 0.21.0.

### PR DESCRIPTION
**Pull request by john-gold**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

### CODE REVIEW

**Comment by john-gold** [comment]
When running `./xbps-src check elvish` the same tests fail as in the CI pipeline.

But running `go test ./...` locally in the [elvish repository](https://github.com/elves/elvish/tree/v0.21.0) succeeds.

**Comment by john-gold** [comment]
Turns out these unit tests have been enabled with go build style just two months ago (with 332756ed0859dd6a1d2c45532642e1bc55f908c4). Before that commit they have never been executed with this template.

I locally ran `xbps-src check elvish` with enabled unit tests in previous versions like 0.20.1 (2a2940e317d99b2e3c9505e936b6ad80d5e5a2a1) and the same tests are failing. Therefore, I suspect that this problem has not been noticed until now because the tests were deactivated.

I suggest disabling the tests temporarily until a solution to the issue is found (as mentioned the test fail only when run via xbps-src.

File: srcpkgs/elvish/template [hunk 1]
```diff
@@ -11,12 +11,17 @@ license="BSD-2-Clause"
...
 changelog="https://elv.sh/blog/${version}-release-notes.html"
 distfiles="https://github.com/elves/elvish/archive/v${version}.tar.gz"
-checksum=6a6006015f44def98676eaed611702b000d66838c0e76da572d517d9bde5c388
+checksum=3a4b93c3c99fe2f9847de35d64be24e2d4b9c12d429cd9831b4571993a66bb7a
 register_shell="/bin/elvish"
 
 # fix: collect2: fatal error: cannot find 'ld'
 LDFLAGS="-fuse-ld=bfd"
 
+do_check() {
```
Comment (classabbyamp) [inline srcpkgs/elvish/template:20]: set `make_check=no` instead (see [Manual.md](https://github.com/void-linux/void-packages/blob/master/Manual.md) for more info)

Comment (john-gold) [reply srcpkgs/elvish/template]: okay

File: srcpkgs/elvish/template [hunk 2]
```diff
@@ -11,8 +11,9 @@ license="BSD-2-Clause"
 homepage="https://elv.sh/"
 changelog="https://elv.sh/blog/${version}-release-notes.html"
 distfiles="https://github.com/elves/elvish/archive/v${version}.tar.gz"
-checksum=6a6006015f44def98676eaed611702b000d66838c0e76da572d517d9bde5c388
+checksum=3a4b93c3c99fe2f9847de35d64be24e2d4b9c12d429cd9831b4571993a66bb7a
 register_shell="/bin/elvish"
+make_check=no # issues with go unit tests when running via xbps-src
```
Comment (classabbyamp) [inline srcpkgs/elvish/template:16]: the only tests that fail are something that seems to depend on some file structure, so just `make_check_args="-skip TestLocation"`

Comment (john-gold) [reply srcpkgs/elvish/template]: Agree, that is the better solution. I was not aware of the `-skip` parameter for `go test`.

**Comment by john-gold** [comment]
@classabbyamp is anything missing?

**Pull request closed without merging**